Fix deleting toolbar from its own event handler (ticket #18309)

a-wi opened the pull request:
This is the fix for bug [18309](http://trac.wxwidgets.org/ticket/18309).

vadz commented on src/msw/toolbar.cpp:
```diff
@@ -1503,19 +1508,17 @@ bool wxToolBar::MSWCommand(WXUINT WXUNUSED(cmd), WXWORD id_)
         return true;
     }
 
-    gs_liveToolbar = NULL;
```
Shouldn't we call `gs_liveToolbars.erase()` here? I don't think anything catastrophic will happen if we don't, but it seems logically wrong to let a toolbar whose events are not being processed in this hash set.

And it's not like it costs us anything, as we already call `find()` anyhow.

vadz reviewed (approved):
Thanks, this is definitely the right thing to do to fix the behaviour when using "embedded toolbars".

We could actually use a stack here, which doesn't require using ugly macros (we'd use `wxStack<>` directly and not `WX_DECLARE_STACK`) and might be a bit more clear as we really only want to push the pointer during the event handler execution and pop it afterwards, but it's a not a huge difference, so if you prefer this version, it's fine too.

Thanks!

a-wi committed:
Fix deleting "embedded" toolbar from its own event handler

When we create a toolbar from within the event handler of another toolbar several times (so we have a chain of embedded toolbars) and the last toolbar is destroyed, gs_liveToolbar variable is reset in its dtor what prevents from executing the part of code in MSWCommand() for other existing toolbars if they are still in their event handlers too (still executing OnLeftClick() function). This may happen when toolbars are created in the chain of modal dialog boxes.
To prevent this interference between the toolbars from occurring we have to keep track of all of active toolbars in the chain instead of keeping track of only the last one.

Closes #18309.
